### ISSUE TITLE
Parentheses around `infer T` in rest parameter position leads to an error

### ISSUE THREAD

**Issue by uhyo** [microsoft/TypeScript#30502]
**TypeScript Version:**  3.3.4000,  3.4.0-rc, 3.4.0-dev.20190320

**Search Terms:** infer parentheses rest parameter

**Code**

```ts
type F1 = ()=>void;

// No error
type T1 = F1 extends (...args: infer T) => void ? T : never;
// error TS2370: A rest parameter must be of an array type.
type T2 = F1 extends (...args: (infer T)) => void ? T : never;
```

**Expected behavior:**
Both `T1` and `T2` should not cause errors because they only differ in excess parentheses.

**Actual behavior:**
`T1` works fine but `T2` emits an error.

**Playground Link:** https://www.typescriptlang.org/play/#src=type%20F1%20%3D%20()%3D%3Evoid%3B%0A%0A%2F%2F%20No%20error%0Atype%20T1%20%3D%20F1%20extends%20(...args%3A%20infer%20T)%20%3D%3E%20void%20%3F%20T%20%3A%20never%3B%0A%2F%2F%20error%20TS2370%3A%20A%20rest%20parameter%20must%20be%20of%20an%20array%20type.%0Atype%20T2%20%3D%20F1%20extends%20(...args%3A%20(infer%20T))%20%3D%3E%20void%20%3F%20T%20%3A%20never%3B

**Related Issues:** none

**Comment by xyx-is** [comment]
This problem seems to happen not only on rest parameter, but also on type parameters with a condition.

```typescript
type IsNumber<T extends number> = T;

// No error
type TestCase_Ok<T> = T extends IsNumber<infer N> ? true : false;
// Type 'N' does not satisfy the constraint 'number'.(2344)
type TestCase_Ng<T> = T extends IsNumber<(infer N)> ? true : false;
```
Note: The code formatter Prettier, which is widely used on such as vscode, removes parens of TestCase_Ng as code formatting.

[Playground Link](https://www.typescriptlang.org/play/#code/C4TwDgpgBAkgzgOQK4FsBGEBOAeAKlCAD2AgDsATOKU1DTAPigF4pcBuAWAChuB6XqAgD2BTJiGZuoSKwhxgAYQCGcCAH0A8gGs8jFviIkKVeMnRZsAS1IAzLIMYB+KMExJoALig2lAG1WcXPys4NAA5AhhUORCctRCwFBwSsCWcDYgLgAW0ADGQqTymErWiWE05phhAHQAFABMAMwALM0AlFKhsvLKqmoIAOa6zKwExGSUsIi0FrXWdpiCbU4ubp7efgHcQA)

**Comment by uhyo** [comment]
@xyx-is thank you for your information! It helped me fix the issue.
